akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingQueriesSpec.scala

```diff
@@ -113,15 +112,17 @@ abstract class ClusterShardingQueriesSpec
       runOn(busy, second, third) {
         val probe = TestProbe()
         val region = ClusterSharding(system).shardRegion(shardTypeName)
-        region.tell(ShardRegion.GetClusterShardingStats(10.seconds), probe.ref)
-        val regions = probe.expectMsgType[ShardRegion.ClusterShardingStats].regions
-        regions.size shouldEqual 3
-        val timeouts = numberOfShards / regions.size
-
-        // 3 regions, 2 shards per region, all 2 shards/region were unresponsive
-        // within shard-region-query-timeout, which only on first is 0ms
-        regions.values.map(_.stats.size).sum shouldEqual 4
-        regions.values.map(_.failed.size).sum shouldEqual timeouts
+        awaitAssert {
+          region.tell(ShardRegion.GetClusterShardingStats(10.seconds), probe.ref)
+          val regions = probe.expectMsgType[ShardRegion.ClusterShardingStats].regions
+          regions.size shouldEqual 3
+          val timeouts = numberOfShards / regions.size
+
+          // 3 regions, 2 shards per region, all 2 shards/region were unresponsive
+          // within shard-region-query-timeout, which only on first is 0ms
+          regions.values.map(_.stats.size).sum shouldEqual 4
+          regions.values.map(_.failed.size).sum shouldEqual timeouts
+        }
```

Comment:
you might have to define a longer timeout for the awaitAssert, since the default might be too short for a full rebalance

Reply:
Done, I put to 10.seconds. Is that an exaggeration? I saw that the default is 3s. So thought that 5s was also too short. 🤷